packages/deposit-address/src/get-deposit-address/isomorphic.ts

```diff
@@ -1,16 +1,104 @@
-import type { ChainConfig } from "@axelarjs/api/axelar-config/types";
+import type { ChainConfig } from "@axelarjs/api";
 
 import {
+  generateRandomSalt,
+  getActiveChains,
   getDepositAddressFromAxelarNetwork,
+  unwrappable,
+  validateActiveChains,
   validateAddress,
   validateAsset,
   validateChainIds,
 } from "./helpers";
-import type { GetDepositAddressDependencies, SendOptions } from "./types";
+import type {
+  DepositAddressOptions,
+  DepositNativeUnwrapOptions,
+  DepositNativeWrapOptions,
+  GetDepositAddressDependencies,
+  GetDepositServiceDependencies,
+  GetLinkedDepositAddressDependencies,
+  SendOptions,
+} from "./types";
 
 export async function getDepositAddress(
-  params: SendOptions,
+  params: DepositAddressOptions,
   dependencies: GetDepositAddressDependencies
+) {
+  const {
+    asset,
+    sourceChain,
+    destinationChain,
+    environment,
+    destinationAddress,
+  } = params;
+
+  const chainConfigs = await dependencies.configClient.getChainConfigs(
+    params.environment
+  );
+
+  // we consider undefined asset as a native token asset
+  if (!asset) {
+    // returns the native wrap deposit address
+    return getNativeWrapDepositAddress(
+      {
+        destinationAddress,
+        sourceChain,
+        destinationChain,
+        environment,
+        salt: params.options?.salt,
+        refundAddress: params.options?.refundAddress || destinationAddress,
+      },
+      dependencies
+    );
+  } else {
+    // this is an erc20 token, we need to check if it is unwrappable at the destination chain
+    const shouldUnwrapToken = unwrappable(
```

Comment:
we should still give the user the option of receiving the erc20 on the destination chain, i.e. not always unwrapping